### PR TITLE
Support for multiple publishers

### PR DESCRIPTION
**Pull request by kareemk**

### CODE REVIEW

File: lib/promiscuous/dependency.rb [hunk 1]
```diff
@@ -23,6 +24,11 @@ def initialize(*args)
         @hash = @internal_key
       end
     end
```
Comment (nviennot) [inline lib/promiscuous/dependency.rb:26]: add new line

File: lib/promiscuous/dependency.rb [hunk 2]
```diff
@@ -23,6 +24,11 @@ def initialize(*args)
...
+    if @publisher_app
```
Comment (nviennot) [inline lib/promiscuous/dependency.rb]: When is `@publisher_app` nil ?

File: lib/promiscuous/dependency.rb [hunk 3]
```diff
@@ -50,6 +56,7 @@ def as_json(options={})
 
   def self.parse(payload, options={})
     case payload
+    when /^(.+):^(.+):([0-9]+)$/ then new($2, options).tap { |d| d.version = $3.to_i }
```
Comment (nviennot) [inline lib/promiscuous/dependency.rb]: remove this line, it should never match (we never send the app name in the dep)

besides the `^` in the middle would make it never match

File: lib/promiscuous/dependency.rb [hunk 4]
```diff
@@ -6,6 +6,7 @@ class Promiscuous::Dependency
   def initialize(*args)
     options = args.extract_options!
     @type = options[:type]
+    @publisher_app = options[:publisher_app]
```
Comment (kareemk) [inline lib/promiscuous/dependency.rb]: I think I like :from better as param name

File: lib/promiscuous/publisher/model/base.rb [hunk 5]
```diff
@@ -87,10 +87,14 @@ def publish(*args, &block)
 
       # TODO reject invalid options
 
-      if attributes.present? && self.publish_to && options[:to] && self.publish_to != options[:to]
-        raise 'versionned publishing is not supported yet'
+      if attributes.present? && self.publish_to && options[:to] && self.publish_to.scan(/[^\/]+$/).first != options[:to]
```
Comment (nviennot) [inline lib/promiscuous/publisher/model/base.rb]: `self.publish_to != "#{Promiscuous::Config.app}/#{options[:to]}"`

File: lib/promiscuous/dependency.rb (hunk 2 above)
Comment (kareemk) [inline lib/promiscuous/dependency.rb]: When used by the publisher. Only the subscriber namespaces.

File: lib/promiscuous/dependency.rb (hunk 3 above)
Comment (kareemk) [inline lib/promiscuous/dependency.rb]: good catch. this was old.

File: lib/promiscuous/subscriber/model/base.rb [hunk 6]
```diff
@@ -46,12 +46,16 @@ def subscribe(*args)
...
 
-      if attributes.present? && self.subscribe_from && options[:from] && self.subscribe_from != options[:from]
+      if attributes.present? && self.subscribe_from && options[:from] && self.subscribe_from.scan(/[^\/]+$/).first != options[:from]
         raise 'Subscribing from different publishers is not supported yet'
       end
 
       unless self.subscribe_from
-        self.subscribe_from = options[:from] || ".*/#{self.name.underscore}"
+        from = options[:from] || self.name.underscore
+        unless from =~ /^__promiscuous__/
```
Comment (nviennot) [inline lib/promiscuous/subscriber/model/base.rb:55]: Maybe adding a level of indirection would have been easier:

Rename `subscribe_from` to `__subscribe_from`

Then

``` ruby
def subscribe_from(options)
  __subscribe_from(options.merge(:from => "#{Promiscuous::Config.app}/#{options[:from]}"))
end
```

Comment (nviennot) [inline lib/promiscuous/subscriber/model/base.rb:55]: Then the embedded docs would use __subscribe_from

File: lib/promiscuous/publisher/model/base.rb (hunk 5 above)
Comment (kareemk) [inline lib/promiscuous/publisher/model/base.rb]: that's how i had it :)

Comment (kareemk) [inline lib/promiscuous/publisher/model/base.rb]: agreed easier to read

File: lib/promiscuous/dependency.rb (hunk 4 above)
Comment (nviennot) [inline lib/promiscuous/dependency.rb]: :owner

File: lib/promiscuous/dependency.rb (hunk 2 above)
Comment (nviennot) [inline lib/promiscuous/dependency.rb]: Oh right

**Pull request closed without merging**